packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -0,0 +1,540 @@
+          this.selectItem(this.indexKeyMap[0]);
+        }
+
+        e.preventDefault();
+      } else if (!this.tagSelectionModel.hasSelection()) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.DOWN]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectNext();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectFirst();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.UP]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (isOpen) {
+        this.focusSelectionModel.selectPrevious();
+        e.preventDefault();
+      } else {
+        this.focusSelectionModel.selectLast();
+        e.preventDefault();
+      }
+    },
+    [KEY_CODES.SPACE]: e => {
+      const { isOpen } = this.getControlledState();
+
+      if (!isOpen) {
+        this.openDropdown();
+        e.preventDefault();
+      }
+    }
+  };
+
+  getItemId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--item-${key}`;
+
+  getTagId = key =>
+    typeof key === 'undefined' ? '' : `${this.getControlledState().id}--tag-${key}`;
+
+  getMenuId = () => `${this.getControlledState().id}--menu`;
+
+  getInputProps = ({ tabIndex = 0, onChange, onKeyDown, onBlur, ...other } = {}) => {
+    const { focusedKey, tagFocusedKey, isOpen } = this.getControlledState();
+
+    return {
+      tabIndex,
+      'aria-autocomplete': 'list',
+      'aria-activedescendant': isOpen ? this.getItemId(focusedKey) : this.getTagId(tagFocusedKey),
+      autoComplete: 'off',
+      onBlur: composeEventHandlers(onBlur, () => {
+        if (!this.menuMousedDown && !this.triggerMousedDown) {
+          this.closeDropdown();
+        }
+      }),
+      onChange: composeEventHandlers(onChange, () => {
+        this.focusSelectionModel.clearSelection();
+      }),
+      onKeyDown: composeEventHandlers(onKeyDown, e => {
+        const keydownHandler = this.INPUT_KEYDOWN_HANDLERS[e.keyCode];
+
+        keydownHandler && keydownHandler(e);
+      }),
+      ...other
+    };
+  };
+
+  getTriggerProps = ({ onMouseDown, onMouseUp, onClick, ...other }) => {
+    return {
+      onMouseDown: composeEventHandlers(onMouseDown, () => {
+        this.triggerMousedDown = true;
+      }),
+      onMouseUp: composeEventHandlers(onMouseUp, () => {
+        this.triggerMousedDown = false;
+      }),
+      onClick: composeEventHandlers(onClick, () => {
+        this.openDropdown();
+        this.inputRef.focus();
+      }),
+      ...other
+    };
+  };
+
+  getMenuProps = ({ onMouseDown, onMouseUp, ...other } = {}) => {
+    return {
+      onMouseDown: composeEventHandlers(onMouseDown, () => {
+        this.menuMousedDown = true;
+      }),
+      onMouseUp: composeEventHandlers(onMouseUp, () => {
+        this.menuMousedDown = false;
+        this.inputRef.focus();
+      }),
+      ...other
```

Comment:
Same here for the above changes we need an `id` & `role` applied

```
id: this.getMenuId(),
role: 'listbox'
```

Reply:
Now added.